nodejs/awsx/lb/loadBalancer.ts

```diff
@@ -30,6 +30,20 @@ export abstract class LoadBalancer extends pulumi.ComponentResource {
     constructor(type: string, name: string, args: LoadBalancerArgs, opts: pulumi.ComponentResourceOptions) {
         super(type, name, {}, opts);
 
+        if (args.loadBalancer !== undefined) {
+            if (Object.keys(args).length > 1) {
```

Comment:
I don't think this check will work as intended because the concrete classes (`ApplicationLoadBalancer` and `NetworkLoadBalancer`) pass in a `loadBalancerType` to this constructor.

Reply:
Yep. And `ApplicationLoadBalancer` also passes in default values for `vpc` and `securityGroups`. Arguably those could be removed from `ApplicationLoadBalancer` and we could rely on the defaults here. You may in that case want to set the description in the `securityGroups` here so that we don't lose the description being set in `ApplicationLoadBalancer` by passing in something like `` description: `Default security group for ${this.constructor.name}: ${name}` ``